gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -577,10 +628,13 @@ public String toString() {
    */
   @SuppressWarnings("unchecked")
   private void processRecord(Object convertedRecord, ForkOperator forkOperator, RowLevelPolicyChecker rowChecker,
-      RowLevelPolicyCheckResults rowResults, int branches, CheckpointableWatermark watermark)
+      RowLevelPolicyCheckResults rowResults, int branches, AcknowledgableWatermark watermark)
       throws Exception {
     // Skip the record if quality checking fails
     if (!rowChecker.executePolicies(convertedRecord, rowResults)) {
+      if (watermark != null) {
+        watermark.ack();
+      }
```

Comment:
not sure I understand... I am acking the watermark when the rowCheck returns false here. So I think I am handling this case.